.pre-commit-config.yaml

```
@@ -0,0 +1,44 @@
+# See https://pre-commit.com for more information
+repos:
+  - repo: local
+    hooks:
+      - id: fmt
+        name: fmt
+        description: Format rust files.
+        entry: cargo +nightly fmt
+        language: system
+        types: [rust]
+        args: ["--all", "--"]
```

Adding the `--check` flag shows a more detailed error message.
```suggestion
        args: ["--all", "--", "--check"]
```